.github/workflows/backend.yml

```diff
@@ -325,6 +325,7 @@ jobs:
 
       - run: |
           # mypy does not have granular codes so don't allow specific messages to regress
+          set -euo pipefail
```

Comment:
would be even better to put this at top level everywhere? i can slap this everywhere if you want

```
defaults:
  run:
    shell: bash -euo pipefail {0}
```

Reply:
it probably should -- but I wasn't ready to do that big of a change in this PR